packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -2367,9 +2367,26 @@ export class AdapterClass extends EventEmitter {
         return this._updateConfig({ newConfig });
     }
 
-    private async _updateConfig(options: InternalUpdateConfigOptions): ioBroker.SetObjectPromise {
+    private async _updateConfig(options: InternalUpdateConfigOptions, useAutoEncryption?: boolean): ioBroker.SetObjectPromise {
         // merge the old and new configuration
-        const _config = Object.assign({}, this.config, options.newConfig);
+        let _config = Object.assign({}, this.config, options.newConfig);
```

Comment:
I would use extendobject and not the stored data to prevent timing issues.

Reply:
Do you mean I must use _config in the rest of the method, if this.config changes again, before the method ends?

Reply:
The problem is that this.confoig COULD be outdated if an user does stuff in parallel but the change did not yet reached the adapter instance.

To prevent edge cases and flow things I would no use this.config at all. I would just check if in the newCOnfig there are fields that needs to be encrypted, do that and then use extendObject to have "one atomic write" to the DB. Because with this alwys the DB value is the relevant one and this.config is updated via an other subscribe mechanism - or better an adapter restart will happen anyway if it changes.